### PR TITLE
#82 Add CI/CD Workflow

### PR DESCRIPTION
**Pull request by apache8080**
Handles #82

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM ubuntu:18.04
+
+COPY . $HOME/ILLIXR/
```
Comment (mhuzai) [inline Dockerfile]: How does this actually work? Does the GitHub server always puts ILLIXR in $HOME/ ?

File: install_deps.sh [hunk 2]
```diff
@@ -73,37 +55,20 @@ then
...
-			-D CMAKE_INSTALL_PREFIX=install
-		sudo make -C "${temp_dir}/Vulkan-Headers/build" "-j$(nproc)" install
+		./scripts/install_vulkan_headers.sh
 	fi
 
 	if [ ! -d "${opt_dir}/googletest" ] && y_or_n "Next: Install gtest"; then
-		git clone https://github.com/google/googletest --branch release-1.10.0 "${opt_dir}/googletest"
-		cmake -S "${opt_dir}/googletest" -B "${opt_dir}/googletest/build"
-		make -C "${opt_dir}/googletest/build" "-j$(nproc)"
+		./scripts/install_gtest.sh
```
Comment (mhuzai) [inline install_deps.sh:71]: How are `opt_dir` and `temp_dir` being passed on to the child scripts?

File: install_deps.sh [hunk 3]
```diff
@@ -119,26 +84,20 @@ then
 	# fi
 
 	if [ ! -d "${temp_dir}/OpenXR-SDK" ] && y_or_n "Next: Install OpenXR SDK from souce"; then
-		git clone https://github.com/KhronosGroup/OpenXR-SDK.git "${temp_dir}/OpenXR-SDK"
-		cmake -S "${temp_dir}/OpenXR-SDK" -B "${temp_dir}/OpenXR-SDK/build"
-		sudo make -C "${temp_dir}/OpenXR-SDK/build" "-j$(nproc)" install
+		./scripts/install_openxr.sh
```
Comment (mhuzai) [inline install_deps.sh:87]: Same question as above about passing `temp_dir` in.

File: install_deps.sh (hunk 2 above)
Comment (apache8080) [reply install_deps.sh]: These files now get sourced so the variables get passed through.

File: Dockerfile (hunk 1 above)
Comment (apache8080) [reply Dockerfile]: The github server does a checkout of the PRs branch and then this line in the dockerfile copies that checked out code into the image.

**Comment by mhuzai** [comment]
Why is the test still failing?

File: scripts/install_openxr.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+git clone https://github.com/KhronosGroup/OpenXR-SDK.git "${temp_dir}/OpenXR-SDK"
+echo "TEST"
```
Comment (mhuzai) [inline scripts/install_openxr.sh:4]:
```suggestion
```

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -1,25 +1,41 @@
...
+
+      - name: Get Branch Name
+        run:  echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
+
+      - name: Login to ghcr.io
+        run: echo ${{ secrets.GHCR_TOKEN }} | docker login ghcr.io -u illixr-ci --password-stdin
+
+      - name: Pull Docker Cache
+        run: docker pull --all-tags ghcr.io/illixr/illixr-tests || true
```
Comment (mhuzai) [inline .github/workflows/main.yaml:25]: What is the `true` for?

**Review by mhuzai — COMMENTED** [review]
`main.yaml`'s future users would greatly benefit from comments. You can figure out what is going on by reading the "name:" fields, but actual comments describing what is going on would be great.

File: .github/workflows/main.yaml (hunk 5 above)
Comment (charmoniumQ) [reply .github/workflows/main.yaml:25]: If pulling fails, we can still build this package (without the benefit of the cache).

**Comment by charmoniumQ** [comment]
> Why is the test still failing?

I am guessing because GitHub does not init/update submodules.

**Review by charmoniumQ — APPROVED** [review]

**Review by mhuzai — APPROVED** [review]

**Pull request merged**